api/src/main/java/org/apache/iceberg/PartitionKey.java

```diff
@@ -98,11 +98,12 @@ public String toPath() {
    * @param row a StructLike row
    */
   @SuppressWarnings("unchecked")
-  public void partition(StructLike row) {
+  public PartitionKey partition(StructLike row) {
```

Comment:
I'm think this will produce an api compatibility issue,  why do we need to change this basic API ?

Reply:
This is just sidecar change, I made it to align with other `StructLike` wrapper, `StructProjection.wrap`, `IndexedStructLike.wrap`, `InternalRowWrapper.wrap`,... to name a few.
@openinx Could you tell me why this can make an api compatibility issue?

Reply:
Yeah. Let's me roll it back.
@openinx could you help me review other changes

Reply:
Yes, I'm currently checking the whole write path.  I think I will need one or two day to understand the whole newly introduced writers since 0.13.x.  Replacing the old writer API with the new one is a great thing. I think we can collaborate to make this forward. Thanks